counsel.el

```diff
@@ -1438,22 +1438,48 @@ This function should set `ivy--old-re'."
 
 (defun counsel-git-grep-action (x)
   "Go to occurrence X in current Git repository."
-  (when (string-match "\\`\\(.*?\\):\\([0-9]+\\):\\(.*\\)\\'" x)
-    (let ((file-name (match-string-no-properties 1 x))
-          (line-number (match-string-no-properties 2 x)))
+  (let ((file-and-line-number (counsel--git-grep-file-and-line-number x)))
+    (when file-and-line-number
       (find-file (expand-file-name
-                  file-name
+                  (car file-and-line-number)
                   (ivy-state-directory ivy-last)))
-      (goto-char (point-min))
-      (forward-line (1- (string-to-number line-number)))
-      (when (re-search-forward (ivy--regex ivy-text t) (line-end-position) t)
-        (when swiper-goto-start-of-match
-          (goto-char (match-beginning 0))))
-      (swiper--ensure-visible)
-      (run-hooks 'counsel-grep-post-action-hook)
-      (unless (eq ivy-exit 'done)
-        (swiper--cleanup)
-        (swiper--add-overlays (ivy--regex ivy-text))))))
+      (counsel--git-grep-go-to-location (cdr file-and-line-number)))))
+
+(defun counsel-git-grep-action-other-window (x)
+  "Go to occurrence X in current Git repository in another window."
+  (let ((file-and-line-number (counsel--git-grep-file-and-line-number x)))
+    (when file-and-line-number
+      (find-file-other-window (expand-file-name
+                               (car file-and-line-number)
+                               (ivy-state-directory ivy-last)))
+      (counsel--git-grep-go-to-location (cdr file-and-line-number)))))
+
+(rx-define grouped-regex-matches
+  (seq string-start (group (zero-or-more any))
+       ":" (group (+ digit)) ":"
+       (group (* any)) string-end))
+
+(defun counsel--git-grep-file-and-line-number (x)
+  "Returns a cons cell of file-name and line number"
+  (when (string-match (rx grouped-regex-matches) x)
+    (cons (match-string-no-properties 1 x) (match-string-no-properties 2 x))))
+
+(defun counsel--git-grep-go-to-location (line-number)
+  "Go to line-number within current buffer"
```

Comment:
The first line of a docstring is a short but complete sentence (ending in a full stop), and argument names are written in all-caps.  If you're interested in the full details of Elisp docstring conventions, feel free to look through [`(elisp) Documentation Tips`](https://gnu.org/s/emacs/manual/html_node/elisp/Documentation-Tips.html).

So in this case, the docstring could be something like:
```el
"Go to LINE-NUMBER in current buffer for `counsel-git-grep'."
```

Similarly `counsel--git-grep-file-and-line-number` above could be something like:
```el
  "Extract file name and line number from `counsel-git-grep' line X.
Return a pair (FILE . LINE) on success; nil otherwise."
```